### PR TITLE
Adds a new space prefab, the candy shop, fixes a couple minor bugs

### PR DESCRIPTION
**Pull request by Camryn-Buttes**
[MAPPING] [FEATURE] [BUG]
## About the PR 

Adds a new prefab to the pool of possible space prefabs, the candy shop.
Comes complete with really smooth vibes, an open-ended mystery, and some clues that just don't quite add-up.
There are no wrong answers! But there IS a lot of candy.

Also fixes a bug with floor lamps not... existing, really.

Also fixes another bug where the old path for icing tubes was still being used on the drug den prefab

## Why's this needed? 

Mystery! Candy! Bugfixes!

## Changelog

```changelog
(u)Camryn Buttes
(*)Added a new prefab to the pool of space prefabs: the candy shop! Complete with an open-ended mystery and some clues that don't quite add up. (also lots of candy)
```

### CODE REVIEW

**Comment by Camryn-Buttes** [comment]
UGH, I FORGOT TO TOGGLE TGM FORMAT ON.

I'll save it in the proper format tomorrow.

**Comment by Camryn-Buttes** [comment]
There we go, converted.

File: code/modules/worldgen/prefabs.dm [hunk 1]
```diff
@@ -179,6 +179,14 @@ ABSTRACT_TYPE(/datum/generatorPrefab)
 		prefabPath = "assets/maps/prefabs/prefab_von_ricken.dmm"
 		prefabSizeX = 42
 		prefabSizeY = 40
+
+	candy_shop //Ryn's store from out of time and out of place
+		maxNum = 1
+		probability = 60 //for initial release, reset to 30 or 25 after a week or so
```
Comment (geneva-breaker) [inline code/modules/worldgen/prefabs.dm:185]: please just set this to an appropriate prob, having launched several prefabs this won't negatively impact hype.

**Comment by Sovexe** [comment]
you need to set `rand_pos = 0` on these if you want them to look like you have them in the map editor 
image

**Comment by Camryn-Buttes** [comment]
> you need to set `rand_pos = 0` on these if you want them to look like you have them in the map editor
> 
> image

I was *wondering* how to do that!

Thanks!

File: code/modules/worldgen/prefabs.dm (hunk 1 above)
Comment (Camryn-Buttes) [reply code/modules/worldgen/prefabs.dm:185]: Will do

Comment (Camryn-Buttes) [reply code/modules/worldgen/prefabs.dm]: Done

**Comment by Camryn-Buttes** [comment]
there we go, all cleaned up. It shouldn't be so goshdang messy on the tables anymore now

File: code/obj/item/storage/secure_safe.dm [hunk 2]
```diff
@@ -707,6 +708,13 @@
 	/obj/item/chilly_orb, // a thing to confuse people
 	/obj/item/spacecash/thousand = 3)
 
+/obj/item/storage/secure/ssafe/candy_shop
+	configure_mode = 0
+	random_code = 1
+	spawn_contents = list(/obj/item/robot_foodsynthesizer,\
```
Comment (ZeWaka) [inline code/obj/item/storage/secure_safe.dm:714]: not a fan of the food synth being available for humans, it's for cyborgs for a reason

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/item/storage/secure_safe.dm (hunk 2 above)
Comment (Camryn-Buttes) [reply code/obj/item/storage/secure_safe.dm:714]: The food synth is already available as a spacebux purchase.

It's already not that uncommon to see in human hands.

Comment (Camryn-Buttes) [reply code/obj/item/storage/secure_safe.dm:714]: If that's not cool, do you have any suggestions for replacements?

Comment (Camryn-Buttes) [reply code/obj/item/storage/secure_safe.dm:714]: I could just remove it if it absolutely needs to go, but I'd rather replace it with something similarly food themed.

**Pull request merged**